kernel/environ.mli

```diff
@@ -60,15 +54,15 @@ end
 
 type named_context_val = private {
   env_named_ctx : Constr.named_context;
-  env_named_map : (Constr.named_declaration * lazy_val) Id.Map.t;
+  env_named_map : Constr.named_declaration Id.Map.t;
   (** Identifier-indexed version of [env_named_ctx] *)
   env_named_idx : Constr.named_declaration Range.t;
   (** Same as env_named_ctx but with a fast-access list. *)
 }
 
 type rel_context_val = private {
   env_rel_ctx : Constr.rel_context;
-  env_rel_map : (Constr.rel_declaration * lazy_val) Range.t;
```

Comment:
BTW the lazy_vals in the rel context are not as vulnerable, since safe_typing always has empty rel_context plugins can't get at them. So maybe we should keep them.

Reply:
Do we even use them? I think that virtually all calls to vm_compute are always performed in an empty rel context since they're in proof mode.

Reply:
That's right. That said, if the bench doesn't show anything serious it's always a plus to get rid of imperative code in the kernel. It would also allow more cleanup in the global environment used by the VM (and thus get rid of some memory leaks).